### PR TITLE
Enhance DiagnosticSource Test readability

### PR DESCRIPTION
**Pull request by tarekgh**
Use the C# 8.0 using statement feature to make the code more readable

### CODE REVIEW

**Comment by tarekgh** [comment]
@stephentoub this PR should be addressing your comment you mentioned here https://github.com/dotnet/runtime/pull/37185#discussion_r433290036

File: src/libraries/System.Diagnostics.DiagnosticSource/tests/ActivitySourceTests.cs [hunk 1]
```diff
@@ -20,67 +20,57 @@ public class ActivitySourceTests : IDisposable
...
+
+                using ActivityListener listener = new ActivityListener
                 {
-                    Assert.False(aSource.HasListeners());
+                    ActivityStarted = activity => Assert.NotNull(activity),
+                    ActivityStopped = activity => Assert.NotNull(activity),
+                    ShouldListenTo = (activitySource) => object.ReferenceEquals(aSource, activitySource),
+                    GetRequestedDataUsingParentId = (ref ActivityCreationOptions<string> activityOptions) => ActivityDataRequest.None,
+                    GetRequestedDataUsingContext = (ref ActivityCreationOptions<ActivityContext> activityOptions) => ActivityDataRequest.None
+                };
```
Comment (stephentoub) [inline src/libraries/System.Diagnostics.DiagnosticSource/tests/ActivitySourceTests.cs:67]: How about:
```C#
using var listener = new ActivityListener();
listener.ActivityStarted = activity => Assert.NotNull(activity);
listener.ActivityStopped = activity => Assert.NotNull(activity);
listener.ShouldListenTo = (activitySource) => object.ReferenceEquals(aSource, activitySource);
listener.GetRequestedDataUsingParentId = (ref ActivityCreationOptions<string> activityOptions) => ActivityDataRequest.None;
listener.GetRequestedDataUsingContext = (ref ActivityCreationOptions<ActivityContext> activityOptions) => ActivityDataRequest.None;
```
?  I realize what you have is valid.  My problem with it is that when I read it, I see the using and immediately think that everything inside the braces is the body of the using, with the instance getting disposed of at the closing brace.  Obviously that's not the case, but it's where 20 years of reading C# leads my mind :)

Comment (tarekgh) [reply src/libraries/System.Diagnostics.DiagnosticSource/tests/ActivitySourceTests.cs]: done. Thanks!

Comment (tarekgh) [reply src/libraries/System.Diagnostics.DiagnosticSource/tests/ActivitySourceTests.cs]: @stephentoub please let me know if you have any more comments.

File: src/libraries/System.Diagnostics.DiagnosticSource/tests/ActivitySourceTests.cs [hunk 2]
```diff
@@ -20,67 +20,55 @@ public class ActivitySourceTests : IDisposable
...
-                    {
-                        // no listeners, we should get null activity.
-                        Assert.Null(a1);
-                        Assert.Equal(Activity.Current, current);
-                    }
-                }
+                Activity current = Activity.Current;
+
+                using Activity a1 = aSource.StartActivity("a1");
+                // no listeners, we should get null activity.
```
Comment (stephentoub) [inline src/libraries/System.Diagnostics.DiagnosticSource/tests/ActivitySourceTests.cs:47]: Nit: might want to move the comment up one line

**Review by stephentoub — APPROVED** [review]

**Pull request merged**